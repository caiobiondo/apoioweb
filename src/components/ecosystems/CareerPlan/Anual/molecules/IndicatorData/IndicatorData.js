import React, { Component } from 'react';
import propTypes from 'prop-types';
import { Icon } from 'natura-ui';
import ReactCSSTransitionGroup from 'react-addons-css-transition-group';

import { PercentageFormat, NumberFormat } from 'utils/numberFormat';
import { IndicatorFields } from 'components/ecosystems/CareerPlan/enums/IndicatorTypes';
import { IndicatorFieldsTypesLabels } from 'components/ecosystems/CareerPlan/enums/IndicatorFields';
import CycleConcept from 'components/ecosystems/CareerPlan/atoms/CycleConcept';

import {
  IndicatorDataWrapper,
  IndicatorDataSort,
  IndicatorDataContent,
  IndicatorDataRow,
  IndicatorFloatContent,
  IndicatorFloatContentClose,
  IndicatorDataSortNumber,
  IndicatorDataRowFeatured,
  IndicatorDataValue,
  IndicatorDataSmallLabel,
  IndicatorDataSmallValue,
} from 'components/ecosystems/CareerPlan/Cycles/molecules/IndicatorData/IndicatorData.styles';

export class IndicatorData extends Component {
  onClick = event => {
    const { onClick, indicatorData, isSimulated } = this.props;

    if (isSimulated) {
      return onClick(indicatorData);
    }

    event.preventDefault();
  };

  onClose = event => {
    const { onClose } = this.props;
    event.stopPropagation();

    return onClose();
  };

  renderContent = () => {
    if (!this.props.indicatorType) {
      return this.renderConsolidatedData();
    }

    return this.renderIndicatorData();
  };

  renderConsolidatedData() {
    const { indicatorData } = this.props;
    const { concept, value } = indicatorData;

    return (
      <div>
        <IndicatorDataRowFeatured>
          <IndicatorDataValue>
            <PercentageFormat value={value} isPercentage />
          </IndicatorDataValue>
        </IndicatorDataRowFeatured>
        <IndicatorDataRow>
          <CycleConcept concept={concept} size="large" />
        </IndicatorDataRow>
      </div>
    );
  }

  renderIndicatorData() {
    const { indicatorData, indicatorType } = this.props;
    const { concept, value } = indicatorData.overcoming ? indicatorData.overcoming : {};

    return (
      <div>
        <IndicatorDataRowFeatured>
          <IndicatorDataValue>
            <PercentageFormat value={value} isPercentage />
          </IndicatorDataValue>
        </IndicatorDataRowFeatured>
        <IndicatorDataRow>
          <CycleConcept concept={concept} size="large" />
        </IndicatorDataRow>

        <IndicatorDataRow>
          <IndicatorDataSmallLabel>Objetivo</IndicatorDataSmallLabel>
          <IndicatorDataSmallValue>
            <NumberFormat value={indicatorData.objective} />
          </IndicatorDataSmallValue>
        </IndicatorDataRow>

        {IndicatorFields[indicatorType].map(field => (
          <IndicatorDataRow key={field}>
            <IndicatorDataSmallLabel>{IndicatorFieldsTypesLabels[field]}</IndicatorDataSmallLabel>
            <IndicatorDataSmallValue>
              <NumberFormat value={indicatorData[field]} />
            </IndicatorDataSmallValue>
          </IndicatorDataRow>
        ))}
      </div>
    );
  }

  render() {
    const { indicatorType, indicatorData, setRef, showDetails, isSimulated } = this.props;
    const { concept } = !this.props.indicatorType
      ? indicatorData
      : indicatorData.overcoming ? indicatorData.overcoming : {};

    return (
      <IndicatorDataWrapper
        innerRef={setRef}
        showDetails={showDetails}
        editable={isSimulated}
        onClick={this.onClick}
        size="5"
        bordered
        hasChart
        indicatorType={indicatorType}
      >
        <IndicatorDataSort index={indicatorData.cycle}>
          <NumberFormat value={indicatorData.cycle} showLastDigits={2} />
        </IndicatorDataSort>

        <IndicatorDataContent>
          <IndicatorDataRow>
            <CycleConcept concept={concept} size="large" />
          </IndicatorDataRow>
        </IndicatorDataContent>

        <ReactCSSTransitionGroup
          transitionName="fadeIn"
          transitionAppear
          transitionLeave
          transitionEnter
          transitionEnterTimeout={200}
          transitionAppearTimeout={200}
          transitionLeaveTimeout={200}
        >
          {showDetails && (
            <IndicatorFloatContent>
              <IndicatorFloatContentClose onClick={this.onClose}>
                <Icon file="ico_times" />
              </IndicatorFloatContentClose>

              <IndicatorDataSortNumber>
                <NumberFormat value={indicatorData.cycle} showLastDigits={2} />
              </IndicatorDataSortNumber>
              {this.renderContent()}
            </IndicatorFloatContent>
          )}
        </ReactCSSTransitionGroup>
      </IndicatorDataWrapper>
    );
  }
}

IndicatorData.propTypes = {
  indicatorData: propTypes.object.isRequired,
  setRef: propTypes.func.isRequired,
  showDetails: propTypes.bool.isRequired,
};

export default IndicatorData;
