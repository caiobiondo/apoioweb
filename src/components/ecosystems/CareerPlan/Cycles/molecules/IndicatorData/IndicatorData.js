import React, { Component } from 'react';
import { Popover, Menu } from 'material-ui';
import { injectIntl, FormattedMessage } from 'react-intl';
import IndicatorDataForm from '../../molecules/IndicatorDataForm';

import { IndicatorFields } from 'components/ecosystems/CareerPlan/enums/IndicatorTypes';
import { PercentageFormat, NumberFormat } from 'utils/numberFormat';

import {
  IndicatorDataWrapper,
  IndicatorDataSort,
  IndicatorDataSortCurrent,
  IndicatorDataContent,
  IndicatorDataRow,
  IndicatorDataRowFeatured,
  IndicatorDataRowObj,
  IndicatorDataValue,
  IndicatorDataConceptValue,
  PopoverStyles,
  PopoverContent,
} from './IndicatorData.styles';

export class IndicatorData extends Component {
  state = {};

  onClick = event => {
    const { onClick, indicatorData } = this.props;

    if (this.canFill()) {
      return onClick(indicatorData);
    }

    event.preventDefault();
    this.showPopover();
  };

  setNode = node => {
    this.indicatorDataNode = node;
  };

  canFill = () => {
    const { indicatorData, canFill } = this.props;
    return !indicatorData.isClosed && canFill;
  };

  hidePopover = () => {
    this.setState({ showPopover: false });
  };

  showPopover = () => {
    this.setState({ showPopover: true });
  };

  renderDisabled() {
    const { indicatorData, indicator } = this.props;
    const { value, concept } = indicatorData.overcoming ? indicatorData.overcoming : {};
    const { objective } = indicatorData;

    return (
      <IndicatorDataContent>
        <IndicatorDataRowObj>
          <IndicatorDataValue>
            <NumberFormat value={objective} />
          </IndicatorDataValue>
        </IndicatorDataRowObj>

        {IndicatorFields[indicator.indicatorType].map(field => (
          <IndicatorDataRow key={field}>
            <IndicatorDataValue>
              <NumberFormat value={indicatorData[field]} />
            </IndicatorDataValue>
          </IndicatorDataRow>
        ))}

        <IndicatorDataRowFeatured>
          <IndicatorDataValue>
            <PercentageFormat value={value} />
          </IndicatorDataValue>
        </IndicatorDataRowFeatured>
        <IndicatorDataRow>
          <IndicatorDataConceptValue concept={concept} />
        </IndicatorDataRow>
      </IndicatorDataContent>
    );
  }

  renderPopover() {
    const { indicatorData } = this.props;

    if (indicatorData.isClosed) {
      return null;
    }

    return (
      <Popover
        open={this.state.showPopover}
        anchorEl={this.indicatorDataNode}
        className="Popover"
        anchorOrigin={{ horizontal: 'middle', vertical: 'center' }}
        targetOrigin={{ horizontal: 'middle', vertical: 'center' }}
        onRequestClose={this.hidePopover}
        style={PopoverStyles}
      >
        <Menu>
          <PopoverContent>
            <FormattedMessage id="careerPlanNotFilled" />
          </PopoverContent>
        </Menu>
      </Popover>
    );
  }

  renderContentNode() {
    const { indicatorData, activeCycle } = this.props;

    if (indicatorData.isClosed) {
      return this.renderDisabled();
    }

    return <IndicatorDataForm {...this.props} isActive={indicatorData.cycle === activeCycle} />;
  }

  renderCurrentLabel() {
    const { indicatorData } = this.props;

    if (!indicatorData.current) {
      return;
    }

    return (
      <IndicatorDataSortCurrent>
        <FormattedMessage id="current" />
      </IndicatorDataSortCurrent>
    );
  }

  render() {
    const { indicator, indicatorData, activeCycle } = this.props;

    return (
      <IndicatorDataWrapper
        indicatorType={indicator.indicatorType}
        editable={!indicatorData.isClosed}
        isActive={indicatorData.cycle === activeCycle}
        key={indicatorData.cycle}
        onClick={this.onClick}
        innerRef={this.setNode}
      >
        <IndicatorDataSort index={indicatorData.cycle}>
          {indicatorData.cycle}
          {this.renderCurrentLabel()}
        </IndicatorDataSort>

        {this.renderContentNode()}

        {this.renderPopover()}
      </IndicatorDataWrapper>
    );
  }
}

export const IndicatorWithIntl = injectIntl(IndicatorData);

export default IndicatorWithIntl;
