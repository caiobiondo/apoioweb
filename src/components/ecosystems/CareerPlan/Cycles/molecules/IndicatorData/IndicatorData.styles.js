import styled from 'styled-components';
import { gray150, gray400, gray700, gray890 } from 'styles/colors';
import { fw600, RobotoRegular } from 'styles/typography';
import * as IndicatorDataFormComponents from '../../molecules/IndicatorDataForm/IndicatorDataForm.styles';
import { IndicatorTypesColors } from 'components/ecosystems/CareerPlan/enums/IndicatorTypes';

export const IndicatorDataContent = IndicatorDataFormComponents.IndicatorDataContent;
export const IndicatorDataRow = IndicatorDataFormComponents.IndicatorDataRow;
export const IndicatorDataRowFeatured = IndicatorDataFormComponents.IndicatorDataRowAcc;
export const IndicatorDataRowObj = IndicatorDataFormComponents.IndicatorDataRowObj;
export const IndicatorDataValue = IndicatorDataFormComponents.IndicatorDataValue;

export const IndicatorDataSortCurrent = styled.span`
  border-radius: 5px;
  color: white;
  font-size: 10px;
  font-weight: ${fw600};
  left: 50%;
  padding: 7px 15px;
  position: absolute;
  text-transform: uppercase;
  top: -25px;
  transform: translate(-50%, 0);
`;

export const IndicatorDataSort = styled.div`
  color: ${gray150};
  font-size: 21px;
  padding: 12px 0;
  position: relative;

  &:before,
  &:after {
    bottom: 0;
    content: '';
    display: block;
    position: absolute;
    top: 0;
    width: 1px;
  }

  &:before {
    background-color: ${gray150};
    right: 0px;
  }

  &:after {
    background-color: white;
    right: -1px;
  }
`;

export const IndicatorFloatContent = styled.div`
  background: white;
  border-radius: 10px;
  bottom: -15px;
  box-shadow: 0 0 25px 0 rgba(0, 0, 0, 0.1);
  left: 50%;
  min-width: 80px;
  position: absolute;
  top: -15px;
  transform: translate(-50%, 0);
  width: 150%;
`;

export const IndicatorFloatContentClose = styled.a`
  background: #fff;
  border-radius: 50%;
  box-shadow: 0 0 7.1px 0.4px rgba(0, 0, 0, 0.1);
  cursor: pointer;
  display: inline-block;
  height: 30px;
  position: absolute;
  right: -10px;
  top: -10px;
  width: 30px;

  svg {
    fill: ${gray700};
    height: 30px;
    width: 10px;
  }
`;

export const IndicatorDataSortNumber = styled.span`
  font-size: 21px;
  padding: 25px 0 20px;
  display: inline-block;
`;

export const IndicatorDataSmallLabel = styled.span`
  font-size: 11px;
  display: block;
  text-transform: uppercase;
`;
export const IndicatorDataSmallValue = styled.span`
  font-size: 11px;
  display: block;
  font-weight: ${fw600};
  margin-bottom: 10px;
`;

export const IndicatorDataWrapper = styled.li`
  cursor: pointer;
  display: inline-block;
  list-style-type: none;
  position: relative;
  text-align: center;
  transition: all 0.2s ease-in;
  transition-property: background, box-shadow, border-radius;
  vertical-align: top;
  white-space: initial;
  width: ${({ size }) => size || '10'}%;
  z-index: 0;

  ${({ isActive }) =>
    isActive &&
    `
    background: ${gray400};
    border-radius: 15px 15px 5px 5px;
    box-shadow: 0px 5px 35px 0 rgba(0, 0, 0, 0.1);
    cursor: default;
    outline: none;
    z-index: 1;

    &:not(:first-child) {
      &:before {
        background-color: #f4f3f3;
        content: '';
        height: 50px;
        left: -2px;
        position: absolute;
        width: 3px;
        z-index: 1;
      }
    }
  `};

  &:last-child {
    ${IndicatorDataSort} {
      &:before,
      &:after {
        display: none;
      }
    }
  }

  &:hover {
    ${({ isActive, editable }) =>
      !isActive &&
      editable &&
      `
      &:after {
        bottom: 0;
        box-shadow: 0px 5px 15px 0 rgba(0,0,0,0.1);
        content: '';
        left: 0;
        position: absolute;
        right: 0;
        top: 50px;
        z-index: -1;
      }
    `};
  }

  ${({ showDetails }) => showDetails && `z-index: 1;`};

  ${({ editable }) => !editable && `cursor: default;`};

  ${({ bordered }) => bordered && `border-right: 1px solid ${gray150}`};

  ${IndicatorDataSort} {
    ${({ isActive }) =>
      isActive &&
      `
      color: ${gray890};

      &:before,
      &:after {
        display: none;
      }
    `};

    ${({ bordered }) =>
      bordered &&
      `
      &:before,
      &:after {
        display: none;
      }
    `};
  }

  ${IndicatorDataSortCurrent} {
    ${({ indicatorType }) => `background-color: ${IndicatorTypesColors[indicatorType]}`};
  }

  ${IndicatorDataSortCurrent} {
    ${({ indicatorType }) => !indicatorType && `background-color: ${gray890}}`};
  }

  ${IndicatorDataSortNumber} {
    ${({ indicatorType }) => `color: ${IndicatorTypesColors[indicatorType]}}`};
  }

  ${IndicatorDataSmallLabel} {
    ${({ indicatorType }) => `color: ${IndicatorTypesColors[indicatorType]}}`};
  }

  ${IndicatorDataFormComponents.IndicatorDataRowInput} {
    ${({ isActive }) => isActive && `box-shadow: 0 0 25px 0 rgba(0, 0, 0, 0.1); `};
  }
`;
