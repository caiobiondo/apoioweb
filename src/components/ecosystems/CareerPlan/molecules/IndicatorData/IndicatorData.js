import React, { Component } from 'react';
import { Popover } from 'material-ui';
import { injectIntl, FormattedMessage } from 'react-intl';
import IndicatorDataForm from '../../molecules/IndicatorDataForm';

import {
  IndicatorDataWrapper,
  IndicatorDataSort,
  IndicatorDataSortCurrent,
  IndicatorDataContent,
  IndicatorDataRow,
  IndicatorDataRowFeatured,
  IndicatorDataRowObj,
  IndicatorDataValue,
  IndicatorDataConceptValue,
  PopoverStyles,
  PopoverContent,
} from './IndicatorData.styles';

export class IndicatorData extends Component {
  state = {};

  onClick = event => {
    const { onClick, indicatorData } = this.props;

    if (this.canFill()) {
      return onClick(indicatorData);
    }

    event.preventDefault();
    this.showPopover();
  };

  setNode = node => {
    this.indicatorDataNode = node;
  };

  canFill = () => {
    const { indicatorData, canFill } = this.props;
    return !indicatorData.isClosed && canFill;
  };

  hidePopover = () => {
    this.setState({ showPopover: false });
  };

  showPopover = () => {
    this.setState({ showPopover: true });
  };

  renderDisabled() {
    const { indicatorData } = this.props;

    return (
      <IndicatorDataContent>
        <IndicatorDataRowObj>
          <IndicatorDataValue>{indicatorData.objective}</IndicatorDataValue>
        </IndicatorDataRowObj>
        <IndicatorDataRow>
          <IndicatorDataValue>{indicatorData.directSale}</IndicatorDataValue>
        </IndicatorDataRow>
        <IndicatorDataRow>
          <IndicatorDataValue>{indicatorData.naturaNetwork}</IndicatorDataValue>
        </IndicatorDataRow>
        <IndicatorDataRowFeatured>
          <IndicatorDataValue>{indicatorData.overcoming}</IndicatorDataValue>
        </IndicatorDataRowFeatured>
        <IndicatorDataRow>
          <IndicatorDataConceptValue concept={indicatorData.consolidated.value} />
        </IndicatorDataRow>
      </IndicatorDataContent>
    );
  }

  renderPopover() {
    const { indicatorData } = this.props;

    if (indicatorData.isClosed) {
      return null;
    }

    return (
      <Popover
        open={this.state.showPopover}
        anchorEl={this.indicatorDataNode}
        className="Popover"
        anchorOrigin={{ horizontal: 'middle', vertical: 'center' }}
        targetOrigin={{ horizontal: 'middle', vertical: 'center' }}
        onRequestClose={this.hidePopover}
        style={PopoverStyles}
      >
        <PopoverContent>
          <FormattedMessage id="careerPlanNotFilled" />
        </PopoverContent>
      </Popover>
    );
  }

  renderContentNode() {
    const { indicatorData, activeCycle } = this.props;

    if (!indicatorData.isClosed) {
      this.renderDisabled();
    }

    return <IndicatorDataForm {...this.props} isActive={indicatorData.cycle === activeCycle} />;
  }

  renderCurrentLabel() {
    const { indicatorData } = this.props;

    if (!indicatorData.current) {
      return;
    }

    return (
      <IndicatorDataSortCurrent>
        <FormattedMessage id="current" />
      </IndicatorDataSortCurrent>
    );
  }

  render() {
    const { indicator, indicatorData, activeCycle } = this.props;

    return (
      <IndicatorDataWrapper
        indicatorType={indicator.indicatorType}
        editable={!indicatorData.isClosed}
        isActive={indicatorData.cycle === activeCycle}
        key={indicatorData.cycle}
        onClick={this.onClick}
        innerRef={this.setNode}
      >
        <IndicatorDataSort index={indicatorData.cycle}>
          {indicatorData.cycle}
          {this.renderCurrentLabel()}
        </IndicatorDataSort>

        {this.renderContentNode()}

        {this.renderPopover()}
      </IndicatorDataWrapper>
    );
  }
}

export const IndicatorWithIntl = injectIntl(IndicatorData);

export default IndicatorWithIntl;
