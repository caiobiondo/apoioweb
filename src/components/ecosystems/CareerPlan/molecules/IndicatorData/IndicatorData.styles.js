import styled, { injectGlobal } from 'styled-components';
import { gray150, blue100, gray400 } from 'styles/colors';
import { fw600 } from 'styles/typography';
import * as IndicatorDataFormComponents from '../../molecules/IndicatorDataForm/IndicatorDataForm.styles';

export const IndicatorDataContent = IndicatorDataFormComponents.IndicatorDataContent;
export const IndicatorDataRow = IndicatorDataFormComponents.IndicatorDataRow;
export const IndicatorDataRowFeatured = IndicatorDataFormComponents.IndicatorDataRowAcc;
export const IndicatorDataRowObj = IndicatorDataFormComponents.IndicatorDataRowObj;
export const IndicatorDataValue = IndicatorDataFormComponents.IndicatorDataValue;

export const IndicatorDataSort = styled.div`
  color: ${gray150};
  font-size: 21px;
  padding: 12px 0;
  position: relative;

  &:before {
    content: '';
    display: block;
    position: absolute;
    top: 0;
    bottom: 0;
    right: -10px;
    width: 1px;
    background-color: ${gray150};
  }

  &:after {
    background-color: white;
    bottom: 0;
    content: '';
    display: block;
    position: absolute;
    right: -11px;
    top: 0;
    width: 1px;
  }
`;

export const IndicatorDataWrapper = styled.li`
  cursor: pointer;
  display: inline-block;
  min-height: 240px;
  list-style-type: none;
  padding: 0 10px;
  position: relative;
  text-align: center;
  transition: all 0.2s ease-in;
  vertical-align: top;
  white-space: initial;
  width: 110px;

  ${IndicatorDataSort} {
    ${({ active }) =>
      active &&
      `
      &:before,
      &:after {
        display: none;
      }
    `};
  }

  ${IndicatorDataFormComponents.IndicatorDataRowInput} {
    ${({ active }) =>
      active &&
      `
        box-shadow: 0 0 25px 0 rgba(0, 0, 0, 0.1);
    `};
  }

  ${({ active }) =>
    active &&
    `
    background: ${gray400};
    box-shadow: 0px 5px 35px 0 rgba(0, 0, 0, 0.1);
    border-radius: 15px 15px 0 0;
    outline: none;
    z-index: 1;

    &:before {
      background-color: #f4f3f3;
      content: '';
      height: 50px;
      left: -2px;
      position: absolute;
      width: 2px;
      z-index: 1;
    }
  `};
`;

export const IndicatorDataSortCurrent = styled.span`
  background-color: ${blue100};
  border-radius: 5px;
  color: white;
  font-size: 10px;
  font-weight: ${fw600};
  left: 50%;
  padding: 7px 15px;
  position: absolute;
  text-transform: uppercase;
  top: -25px;
  transform: translate(-50%, 0);
`;

export const PopoverStyles = {
  color: '#777777',
  fontSize: '14px',
  maxWidth: '250px',
  overflow: 'visible',
  padding: '20px',
};

export const PopoverContent = styled.div`
  position: relative;
  text-align: center;
`;

export const PopoverArrow = styled.div`
  bottom: -60px;
  height: 40px;
  left: 50%;
  overflow: hidden;
  position: absolute;
  transform: translate(-50%, 0);
  width: 100px;

  // &:after {
  //   background: #fff;
  //   box-shadow: -1px -1px 10px 0px rgba(0, 0, 0, 0.5);
  //   content: '';
  //   height: 50px;
  //   left: 25px;
  //   position: absolute;
  //   top: -50px;
  //   transform: rotate(45deg);
  //   width: 50px;
  // }
`;

// Overwrite default behavior for Material-UI popover
injectGlobal`
.Popover > div {
  overflow: visible !important;
}
`;
