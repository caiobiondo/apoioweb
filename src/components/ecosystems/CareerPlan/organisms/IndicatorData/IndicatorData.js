import React, { Component } from 'react';
import { Popover } from 'material-ui';
import IndicatorDataForm from '../../molecules/IndicatorDataForm';

import {
  IndicatorDataWrapper,
  IndicatorDataSort,
  IndicatorDataSortCurrent,
  IndicatorDisabledDataContent,
  IndicatorDisabledDataRow,
  IndicatorDisabledDataRowAcc,
  IndicatorDisabledDataRowObj,
  IndicatorDisabledDataValue,
  PopoverStyles,
  PopoverContent,
  PopoverArrow,
} from './IndicatorData.styles';

export class IndicatorData extends Component {
  state = {};

  onClick = event => {
    const { onClick, indicatorData } = this.props;

    if (!this.canFill()) {
      event.preventDefault();
      this.togglePopover();
      return console.log('Disabled Field');
    }

    return onClick(indicatorData);
  };

  setNode = node => {
    this.indicatorDataNode = node;
  };

  isFake = () => {
    const { indicatorData } = this.props;
    return !indicatorData.preLoaded;
  };

  canFill = () => {
    const { indicatorData, canFill } = this.props;
    return !indicatorData.preLoaded && canFill;
  };

  handleRequestClose = () => {
    this.setState({ open: false });
  };

  togglePopover = () => {
    const { open } = this.state;
    this.setState({ open: !open });
  };

  renderDisabled() {
    const { indicatorData } = this.props;

    return (
      <IndicatorDisabledDataContent>
        <IndicatorDisabledDataRow />
        <IndicatorDisabledDataRowObj>
          <IndicatorDisabledDataValue>{indicatorData.obj}</IndicatorDisabledDataValue>
        </IndicatorDisabledDataRowObj>
        <IndicatorDisabledDataRow>
          <IndicatorDisabledDataValue>{indicatorData.done}</IndicatorDisabledDataValue>
        </IndicatorDisabledDataRow>
        <IndicatorDisabledDataRow>
          <IndicatorDisabledDataValue>{indicatorData.networkDone}</IndicatorDisabledDataValue>
        </IndicatorDisabledDataRow>
        <IndicatorDisabledDataRowAcc>
          <IndicatorDisabledDataValue>
            {indicatorData.accumulatedOverload}
          </IndicatorDisabledDataValue>
        </IndicatorDisabledDataRowAcc>
      </IndicatorDisabledDataContent>
    );
  }

  renderPopover() {
    return (
      <Popover
        open={this.state.open}
        anchorEl={this.indicatorDataNode}
        className="Popover"
        anchorOrigin={{ horizontal: 'middle', vertical: 'center' }}
        targetOrigin={{ horizontal: 'middle', vertical: 'center' }}
        onRequestClose={this.handleRequestClose}
        style={PopoverStyles}
      >
        <PopoverContent>
          O simulador deve ser preenchido sequencialmente. <PopoverArrow />
        </PopoverContent>
      </Popover>
    );
  }

  render() {
    const { indicatorData, index } = this.props;

    const currentNode = indicatorData.current ? (
      <IndicatorDataSortCurrent>Atual</IndicatorDataSortCurrent>
    ) : null;

    const contentNode = this.isFake() ? (
      <IndicatorDataForm {...this.props} />
    ) : (
      this.renderDisabled()
    );

    return (
      <IndicatorDataWrapper
        active={indicatorData.active}
        key={indicatorData.id}
        onClick={this.onClick}
        innerRef={this.setNode}
      >
        <IndicatorDataSort index={index}>
          {index + 1}
          {currentNode}
        </IndicatorDataSort>

        {contentNode}

        {this.renderPopover()}
      </IndicatorDataWrapper>
    );
  }
}

export default IndicatorData;
