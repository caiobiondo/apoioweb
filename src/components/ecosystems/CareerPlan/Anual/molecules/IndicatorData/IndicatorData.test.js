import React from 'react';
import { shallow } from 'enzyme';
import IndicatorData from './IndicatorData';

const setup = propOverrides => {
  const props = Object.assign(
    {
      indicatorData: {
        cycle: 1,
        concept: 'Excede',
      },
      setRef: () => {},
      showDetails: true,
    },
    propOverrides,
  );

  const result = shallow(<IndicatorData {...props} />);

  return {
    props,
    result,
  };
};

describe('IndicatorData', () => {
  describe('onClick method', () => {
    it('should call onClick prop when item was simulated', () => {
      // given
      const eventMock = { preventDefault: jest.fn() };
      const props = {
        indicatorData: { cycle: 1, concept: 'Excede' },
        onClick: jest.fn(),
        isSimulated: true,
      };

      // when
      const { result } = setup(props);
      const instance = result.instance();
      instance.onClick(eventMock);

      // then
      expect(props.onClick).toBeCalledWith(props.indicatorData);
      expect(eventMock.preventDefault).not.toBeCalled();
    });

    it('should not call onClick prop when the item was not simulated', () => {
      // given
      const eventMock = { preventDefault: jest.fn() };
      const props = {
        onClick: jest.fn(),
        isSimulated: false,
      };

      // when
      const { result } = setup(props);
      const instance = result.instance();
      instance.onClick(eventMock);

      // then
      expect(eventMock.preventDefault).toBeCalled();
    });
  });

  describe('onClose method', () => {
    it('should call onClose prop and stopPropagation', () => {
      // given
      const eventMock = { stopPropagation: jest.fn() };
      const props = {
        onClose: jest.fn(),
        isSimulated: true,
      };

      // when
      const { result } = setup(props);
      const instance = result.instance();
      instance.onClose(eventMock);

      // then
      expect(props.onClose).toBeCalled();
      expect(eventMock.stopPropagation).toBeCalled();
    });
  });

  describe('renderContent method', () => {
    it('should call renderConsolidatedData method when a indicatorType is given', () => {
      // given
      const props = {
        indicatorData: {
          cycle: 1,
          overcoming: { concept: 'Excede' },
        },
        indicatorType: 'scoresTotal',
        showDetails: true,
      };

      // when
      const { result } = setup(props);
      const contentNode = result.find('IndicatorDatastyles__IndicatorFloatContent');

      // then
      expect(contentNode).toMatchSnapshot();
    });

    it('should call renderConsolidatedData method when a indicatorType is not given', () => {
      // given
      const props = {
        indicatorData: { cycle: 1, concept: 'Excede' },
        showDetails: true,
      };

      // when
      const { result } = setup(props);
      const contentNode = result.find('IndicatorDatastyles__IndicatorFloatContent');

      // then
      expect(contentNode).toMatchSnapshot();
    });
  });
});
