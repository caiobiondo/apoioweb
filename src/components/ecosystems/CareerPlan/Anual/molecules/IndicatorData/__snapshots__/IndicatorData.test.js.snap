// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndicatorData renderContent method should call renderConsolidatedData method when a indicatorType is given 1`] = `
<IndicatorDatastyles__IndicatorFloatContent>
  <IndicatorDatastyles__IndicatorFloatContentClose
    onClick={[Function]}
  >
    <Icon
      file="ico_times"
    />
  </IndicatorDatastyles__IndicatorFloatContentClose>
  <IndicatorDatastyles__IndicatorDataSortNumber>
    1
  </IndicatorDatastyles__IndicatorDataSortNumber>
</IndicatorDatastyles__IndicatorFloatContent>
`;

exports[`IndicatorData renderContent method should call renderConsolidatedData method when a indicatorType is not given 1`] = `
<IndicatorDatastyles__IndicatorFloatContent>
  <IndicatorDatastyles__IndicatorFloatContentClose
    onClick={[Function]}
  >
    <Icon
      file="ico_times"
    />
  </IndicatorDatastyles__IndicatorFloatContentClose>
  <IndicatorDatastyles__IndicatorDataSortNumber>
    1
  </IndicatorDatastyles__IndicatorDataSortNumber>
</IndicatorDatastyles__IndicatorFloatContent>
`;
